src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,731 @@
+            _verifyOperatorSignature(operator, allocations, operatorSignature);
+        }
+        require(delegation.isOperator(operator), OperatorNotRegistered());
+
+        (bool isSet, uint32 delay) = allocationDelay(operator);
+        require(isSet, UninitializedAllocationDelay());
+
+        // effect timestamp for allocations to take effect. This is configurable by operators
+        uint32 allocationEffectTimestamp = uint32(block.timestamp) + delay;
+        // completable timestamp for deallocations
+        uint32 deallocationCompletableTimestamp = uint32(block.timestamp) + DEALLOCATION_DELAY;
+
+        for (uint256 i = 0; i < allocations.length; ++i) {
+            // 1. For the given (operator,strategy) clear all pending free magnitude for the strategy and update freeMagnitude
+            _updateFreeMagnitude({
+                operator: operator,
+                strategy: allocations[i].strategy,
+                numToComplete: type(uint16).max
+            });
+
+            // 2. Check current totalMagnitude matches expected value. This is to check for slashing race conditions
+            // where an operator gets slashed from an operatorSet and as a result all the configured allocations have larger
+            // proprtional magnitudes relative to each other.
+            uint64 currentTotalMagnitude = _getLatestTotalMagnitude(operator, allocations[i].strategy);
+            require(currentTotalMagnitude == allocations[i].expectedTotalMagnitude, InvalidExpectedTotalMagnitude());
+
+            // 3. set allocations for the strategy after updating freeMagnitude
+            _modifyAllocations({
+                operator: operator,
+                allocation: allocations[i],
+                allocationEffectTimestamp: allocationEffectTimestamp,
+                deallocationCompletableTimestamp: deallocationCompletableTimestamp
+            });
+        }
+    }
+
+    /**
+     * @notice Called by an AVS to slash an operator for given operatorSetId, list of strategies, and bipsToSlash.
+     * For each given (operator, operatorSetId, strategy) tuple, bipsToSlash
+     * bips of the operatorSet's slashable stake allocation will be slashed
+     *
+     * @param operator the address to slash
+     * @param operatorSetId the ID of the operatorSet the operator is being slashed on behalf of
+     * @param strategies the set of strategies to slash
+     * @param bipsToSlash the number of bips to slash, this will be proportional to the
+     * operator's slashable stake allocation for the operatorSet
+     */
+    function slashOperator(
+        address operator,
+        uint32 operatorSetId,
+        IStrategy[] calldata strategies,
+        uint16 bipsToSlash
+    ) external {
+        OperatorSet memory operatorSet = OperatorSet({avs: msg.sender, operatorSetId: operatorSetId});
+        bytes32 operatorSetKey = _encodeOperatorSet(operatorSet);
+        require(isOperatorSlashable(operator, operatorSet), InvalidOperator());
+
```

Comment:
`require(bipsToSlash != 0 && bipsToSlash < BIPS_FACTOR)`

Reply:
Fixed in: #786